MailCore/Cache/MailboxManager/MailboxManager+Thread.swift

```diff
@@ -454,6 +502,15 @@ public extension MailboxManager {
         return thread
     }
 
+    private func upsertMessage(_ message: Message, oldMessage: Message, threadsToUpdate: inout Set<Thread>, using realm: Realm) {
+        keepCacheAttributes(for: message, keepProperties: .standard, using: realm)
+        realm.add(message, update: .modified)
+
+        for thread in oldMessage.threads {
+            threadsToUpdate.insert(thread)
+        }
```

Comment:
I don't know if it's more performant but we can use `formUnion`
```suggestion
        threadsToUpdate.formUnion(oldMessage.threads)
```